graph-builder/src/graph.rs

```diff
@@ -169,6 +186,8 @@ pub fn run<'a>(settings: &'a config::AppSettings, state: &State) -> ! {
 
     // Don't wait on the first iteration
     let mut first_iteration = true;
+    // Don't collect upstream scrape duration on cold start
```

Comment:
I'm wondering if we would prefer to wait until first_success with measuring the scrapes. An unsuccessful first scrape is unlikely to warm up the cache properly.

Reply:
Yes, me too I'd prefer this logic to be linked to `first_success` instead (if possible).

Reply:
`first_success` is being updated way too late if I understood correctly - it happens after graph has been successfully built. 

Updated the code to store after scrape has been successful (`registries` are set, even if empty) - not entirely sure if that is correct though

Reply:
I believe the suggestion was to stop the timer, keep the value locally, and then later set/observe in the correct metrics based on success status.

Reply:
Implemented decision making in `first_success`